tests/generic_config_updater/gu_utils.py

```diff
@@ -286,3 +289,49 @@ def check_vrf_route_for_intf(duthost, vrf_name, intf_name, is_ipv4=True):
     output = duthost.shell("show {} route vrf {} | grep -w {}".format(address_family, vrf_name, intf_name))
 
     pytest_assert(not output['rc'], "Route not found for {} in vrf {}".format(intf_name, vrf_name))
+
+
+def get_gcu_field_operations_conf():
+    conf_path = os.path.join(os.path.dirname(__file__), GCU_FIELD_OPERATION_CONF_FILE)
+    with open(conf_path, 'r') as s:
+        gcu_conf = json.load(s)
+    return gcu_conf
+
+
+def get_asic_name(duthost):
+    asic_type = duthost.facts["asic_type"]
+    asic = "unknown"
+    gcu_conf = get_gcu_field_operations_conf()
+    asic_mapping = gcu_conf["helper_data"]["rdma_config_update_validator"]
+    if asic_type == 'cisco-8000':
+        asic = "cisco-8000"
+    elif asic_type == 'mellanox' or asic_type == 'vs' or asic_type == 'broadcom':
+        hwsku = duthost.shell(GET_HWSKU_CMD)['stdout'].rstrip('\n')
+        if asic_type == 'mellanox' or asic_type == 'vs':
+            spc1_hwskus = asic_mapping["mellanox_asics"]["spc1"]
+            if hwsku.lower() in [spc1_hwsku.lower() for spc1_hwsku in spc1_hwskus]:
+                asic = "spc1"
+                return asic
+        if asic_type == 'broadcom' or asic_type == 'vs':
+            broadcom_asics = asic_mapping["broadcom_asics"]
+            for asic_shorthand, hwskus in broadcom_asics.items():
+                if asic != "unknown":
```

Comment:
>asic [](http://example.com/codeflow?start=19&length=4)

The if-block is not needed. You may consider inner loop with for-else-break.

Reply:
Thanks, changed to for-else-break syntax.